Services/Thoughts.WebAPI/Controllers/BlogPostController.cs

```diff
@@ -0,0 +1,231 @@
+﻿using Microsoft.AspNetCore.Mvc;
+
+using Thoughts.Domain.Base.Entities;
+using Thoughts.Interfaces;
+using Thoughts.Services.Models;
+
+namespace Thoughts.WebAPI.Controllers
+{
+    [ApiController]
+    [Route(WebAPIAddresses.BlogPosts)]
+    public class BlogPostController : ControllerBase
+    {
+        private readonly IBlogPostManager _manager;
+        private readonly ILogger<BlogPostController> _logger;
+
+        public BlogPostController(IBlogPostManager manager, ILogger<BlogPostController> logger)
+        {
+            _manager = manager;
+            _logger = logger;
+        }
+
+        /// <summary>Назначение тэга посту</summary>
+        /// <param name="dto">DTO ключевого слова поста</param>
+        /// <param name="Cancel">Токен отмены</param>
+        /// <returns>Истина, если тэг был назначен успешно</returns>
+        [HttpPut("posts/{PostId}/tag/{Tag}")]
+        public async Task<IActionResult> AssignTagAsync([FromBody] PostTagDTO dto, CancellationToken Cancel = default)
```

Comment:
```suggestion
        [HttpPut("posts/{PostId}/tag/{Tag}")]
        public async Task<IActionResult> AssignTagAsync([FromRoute] PostTagDTO dto, CancellationToken Cancel = default)
```
Если параметры модели указаны в маршруте, то надо указать системе привязки, что брать их надо не из тела запроса, а из параметров маршрута. Это нужно для Post и Put запросов (которые имеют тело)

Reply:
Исправлено